Tests/SharingTests/PublisherTests.swift

```diff
@@ -31,9 +32,89 @@
       #expect(values.withLock(\.self) == [0, 42])
 
       $value = Shared(wrappedValue: 0, .inMemory("count"))
+      #expect(values.withLock(\.self) == [0, 42, 42])
```

Comment:
- [ ] Can we suppress this duplicate emission when replacing with the same reference/object ID?